service/src/main/java/bio/terra/workspace/common/utils/GcpUtils.java

```diff
@@ -81,4 +85,21 @@ public static String getControlPlaneProjectId() {
                 new IllegalStateException(
                     "Could not determine default GCP control plane project ID."));
   }
+
+  /**
+   * Returns the email of the application default credentials, which should represent a service
+   * account in all WSM deployments.
+   */
+  public static String getWsmSaEmail(GoogleCredentials wsmCredentials) {
+    // WSM always runs as a service account, but credentials for that SA may come from different
+    // sources depending on whether it is running in GCP or locally.
+    if (wsmCredentials instanceof ComputeEngineCredentials) {
+      return ((ComputeEngineCredentials) wsmCredentials).getAccount();
```

Comment:
NittyNit: if `getAccount()` is from a super class/interface, then you could do it in a single if.

Reply:
Good point, I didn't realize the two subclasses both implemented the same interface!